crates/client-api/src/routes/database.rs

```diff
@@ -523,15 +507,14 @@ pub async fn sql<S>(
     State(worker_ctx): State<S>,
     Path(SqlParams { name_or_address }): Path<SqlParams>,
     Query(SqlQueryParams {}): Query<SqlQueryParams>,
-    auth: SpacetimeAuthHeader,
+    Extension(auth): Extension<SpacetimeAuth>,
     body: String,
 ) -> axum::response::Result<impl IntoResponse>
 where
     S: NodeDelegate + ControlStateDelegate,
 {
     // Anyone is authorized to execute SQL queries. The SQL engine will determine
     // which queries this identity is allowed to execute against the database.
-    let auth = auth.get().ok_or((StatusCode::UNAUTHORIZED, "Invalid credentials."))?;
```

Comment:
(review note: I believe this used to fail if there was no auth, but it's fine because I think `AuthCtx` does identity checking?)

Reply:
`AuthCtx` doesn't do identity checking, but it actually just didn't make sense to check this before, given the comment right above it. We don't need to require credentials for this.